src/main/java/appeng/fluids/parts/PartFluidStorageBus.java

```diff
@@ -0,0 +1,508 @@
+			// but it does not return an inventory for the action source, we do NOT fall back to using
+			// IItemHandler's, as that might circumvent the security setings, and might also cause
+			// performance issues.
+			return null;
+		}
+
+		// Check via cap for IItemHandler
+		IFluidHandler handlerExt = target.getCapability( CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, targetSide );
+		if( handlerExt != null )
+		{
+			return new FluidHandlerAdapter( handlerExt, this );
+		}
+
+		return null;
+	}
+
+	@Override
+	public TickRateModulation tickingRequest( IGridNode node, int TicksSinceLastCall )
+	{
+		if( this.resetCacheLogic != 0 )
+		{
+			this.resetCache();
+		}
+
+		if( this.monitor != null )
+		{
+			return this.monitor.onTick();
+		}
+
+		return TickRateModulation.SLEEP;
+	}
+
+	protected void resetCache()
+	{
+		final boolean fullReset = this.resetCacheLogic == 2;
+		this.resetCacheLogic = 0;
+
+		final IMEInventory<IAEFluidStack> in = this.getInternalHandler();
+		IItemList<IAEFluidStack> before = AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ).createList();
+		if( in != null )
+		{
+			before = in.getAvailableItems( before );
+		}
+
+		this.cached = false;
+		if( fullReset )
+		{
+			this.handlerHash = 0;
+		}
+
+		final IMEInventory<IAEFluidStack> out = this.getInternalHandler();
+
+		if( in != out )
+		{
+			IItemList<IAEFluidStack> after = AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ).createList();
+			if( out != null )
+			{
+				after = out.getAvailableItems( after );
+			}
+			Platform.postListChanges( before, after, this, this.source );
+		}
+	}
+
+	protected void resetCache( final boolean fullReset )
+	{
+		if( this.getHost() == null || this.getHost().getTile() == null || this.getHost().getTile().getWorld() == null || this.getHost().getTile().getWorld().isRemote )
+		{
+			return;
+		}
+
+		if( fullReset )
+		{
+			this.resetCacheLogic = 2;
+		}
+		else
+		{
+			this.resetCacheLogic = 1;
+		}
+
+		try
+		{
+			this.getProxy().getTick().alertDevice( this.getProxy().getNode() );
+		}
+		catch( final GridAccessException e )
+		{
+			// :P
+		}
+	}
+
+	@Override
+	public boolean onPartActivate( final EntityPlayer player, final EnumHand hand, final Vec3d pos )
+	{
+		if( !player.isSneaking() )
+		{
+			if( Platform.isClient() )
+			{
+				return true;
+			}
+			Platform.openGUI( player, this.getHost().getTile(), this.getSide(), GuiBridge.GUI_STORAGEBUS_FLUID );
+			return true;
+		}
+
+		return false;
+	}
+
+	@Override
+	public void onChangeInventory( final IItemHandler inv, final int slot, final InvOperation mc, final ItemStack removedStack, final ItemStack newStack )
+	{
+		super.onChangeInventory( inv, slot, mc, removedStack, newStack );
+
+		if( inv == this.config )
+		{
+			this.resetCache( true );
+		}
+	}
+
+	@Override
+	public IItemHandler getInventoryByName( final String name )
+	{
+		if( name.equals( "config" ) )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "config" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move the "config" string literal on the left side of this string comparison. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1132)